src/Widgets/HeaderBar.vala

```diff
@@ -88,10 +103,32 @@ public class Camera.Widgets.HeaderBar : Gtk.HeaderBar {
         mode_switch = new Granite.ModeSwitch.from_icon_name (PHOTO_ICON_SYMBOLIC, VIDEO_ICON_SYMBOLIC);
         mode_switch.valign = Gtk.Align.CENTER;
 
+        camera_menu_button = new Gtk.MenuButton ();
+        camera_options = new Gtk.Menu ();
+        camera_menu_button.set_popup (camera_options);
+
+        Gtk.CssProvider camera_menu_button_style_provider = new Gtk.CssProvider ();
+
+        try {
+            camera_menu_button_style_provider.load_from_data (CAMERA_MENU_BUTTON_STYLESHEET, -1);
+        } catch (Error e) {
+            warning ("Styling take button failed: %s", e.message);
+        }
+
+        var camera_menu_button_style_context = camera_menu_button.get_style_context ();
+        camera_menu_button_style_context.add_provider (camera_menu_button_style_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
+        camera_menu_button_style_context.add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
+        camera_menu_button_style_context.add_class ("camera-menu");
+
+        var linked_box = new Gtk.Box (Gtk.Orientation.HORIZONTAL, 0);
```

Comment:
Typically Box isn't preferred anymore. Grid is horizontal by default so this could be 
```suggestion
        var linked_box = new Gtk.Grid ();
```